plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/config/WidgetModel.java

```diff
@@ -439,6 +451,63 @@ public void modelDirtyStateChanged(IStructuredModel model, boolean isDirty) {
 		}
 	}
 
+	private void updateEnginesFromConfigXML() {
```

Comment:
This method looks like something that belongs to _HybridMobileEngineManager_. I'd rather keep WidgetModel about access to the information on config.xml, I do not think we should add logic that modifies the project to it.